src/commands/registry.rs

```diff
@@ -49,6 +49,10 @@ pub struct Push {
     )]
     pub insecure: bool,
 
+    /// Specifies to compose components before pushing the application.
```

Comment:
Maybe help folks along with e.g.

```suggestion
    /// Compose component dependencies before pushing the application.
    ///
    /// The default is to compose before pushing, which maximises compatibility with
    /// different Spin runtime hosts. Turning composition off can optimise
    /// bandwidth for shared dependencies, but makes the pushed image incompatible
    /// with hosts that cannot carry out composition themselves.
```

Wordier I know but maybe helpful?